src/main/cpp/utilities/wavereader.cpp

```diff
@@ -135,30 +162,67 @@ waveFile WaveReader::fileToBuffer( std::string inputFile )
     int sampleSize = header.bitsPerSample / 8;
     int i;
 
-    temp_buffer = ( short* ) malloc( sizeof( short ) * amountOfSamples );
-
-    // Read samples from WAV file
-    for ( i = 0; i < amountOfSamples; ++i )
-        fread( &temp_buffer[ i ], ( size_t ) sampleSize, 1, fp );
```

Comment:
We now write into the AudioBuffer directly when reading data (thus removing an extra loop, see template function _convert_).